src/main/java/org/spongepowered/common/world/extent/ExtentViewDownsize.java

```diff
@@ -458,11 +459,19 @@ public boolean spawnEntity(Entity entity, Cause cause) {
 
     @Override
     public Entity createEntity(EntityType type, Vector3d position) throws IllegalArgumentException, IllegalStateException {
+        checkNotNull(position, "The position cannot be null!");
         checkBlockRange(position.getX(), position.getY(), position.getZ());
         return this.extent.createEntity(type, position);
     }
 
     @Override
+    public Entity createEntityNaturally(EntityType type, Vector3d position) throws IllegalArgumentException, IllegalStateException {
+        checkNotNull(position, "The position cannot be null!");
+        checkBlockRange(position.getX(), position.getY(), position.getZ());
```

Comment:
No.... You don't want to check block range..... Entities can exist outside the bounds of blocks.